### DISCUSSION TITLE
Add a way to reference the parent resource from a sub-resource

### DISCUSSION THREAD

**Discussion by peachey2k2** [godotengine/godot-proposals#8506] [Scripting]
Godot's node system works so well because they have so many ways to communicate with each other. One can for example:
* Access a node at the other side of the node tree with a simple drag and drop
* Get the first child of a node with and move it way down below with `move_child(get_child(0), -1)`
* Free every child of a node just by iterating over `Node.get_children()`
* and so much more...

This system is pretty good, but not having that sort of thing for custom resources is pretty annoying. Yeah, I can just hold a reference to `Node.get_tree()` in a singleton, but that still only works for nodes. At the moment, there is no way that I know of to access from a resource to another that is one level higher without doing some pointlessly complicated coding (like turning the main resource to a tool script and having a setter for the other resource that pssses the reference) where we could just have a function similar to `Node.get_parent()` but for resources.

**Comment by AThousandShips** [comment]
Resources are shared, so there's no *one* parent resource, this wouldn't work except specific cases where exactly one parent exists

More generally this kind of use of resources is not how they are supposed to be used, they aren't supposed to know "where" they are

If you *do* need to know this relationship then what you need is not a `Resource`, see [here](https://docs.godotengine.org/en/stable/tutorials/best_practices/node_alternatives.html)
